Documentation/Benchmark_Network.md

```diff
@@ -0,0 +1,123 @@
+The session layer is responsible for establishing, maintaining, and terminating connections between applications.
+
+<p>
+    <img src="https://jmvidal.cse.sc.edu/talks/javasockets/osiSockets.gif" alt>
+    <em>OSI Model</em>
+</p>
+
+(TCP or UDP is defined in the transport layer)
+
+---
+
+## Definition of the Objectives
+
+The objective of this benchmark is to compare the different network libraries and evaluate possible solutions to create our protocol.
+Specifically, we aim to evaluate the performance, scalability, and usability of various network libraries.
+
+### Goals:
+- **Determine the best combination** of performance, speed, scalability, integration, and usability for our protocol.
+- Ensure the benchmark takes cross-platform compatibility between Windows and Linux into account.
+
+
+---
+
+## Definition of the Metrics
+
+We will use the following metrics to evaluate the performance of the network libraries:
+- **Latency**: The time taken for data to travel from client to server and back.
+- **Bandwidth**: The data transfer rate over a network connection.
+- **CPU Usage**: The amount of processing power used by the networking operations.
+- **Memory Usage**: The amount of system memory consumed.
+- **Network Usage**: The amount of network resources utilized (data sent/received).
+- **Packet Loss**: The rate at which packets fail to reach their destination.
+- **Scalability**: How well the library handles a growing number of clients or data size.
+- **Integration**: How well the library integrates with C++ and other project requirements.
+- **Usability**: The ease of use and learning curve for developers.
+- **(Security)**: Optional metric, depending on the protocol's needs.
+- **(Reliability)**: Optional metric, measuring how well the system maintains connections over time.
+
+---
+
+## Scope of the Benchmarking
+
+The benchmark will focus on the following network libraries:
+- **ASIO (Asio C++ Library)**: An asynchronous I/O library that is cross-platform and designed to be efficient and lightweight.
+- **Winsock (Windows Socket API)**: The primary network interface for Windows systems, optimized for the platform.
+- **Unix BSD Socket (Unix Socket API)**: The standard network interface for Unix-like systems (Linux, macOS, etc.).
+
+**Key Notes**:
+- **ASIO**: A cross-platform library that compiles easily on both Windows and Linux.
+- **Winsock** and **Unix BSD Socket**: Handle networking differently, requiring abstraction layers for full cross-platform compatibility.
+
+---
+
+# Benchmark Results (Placeholder)
+
+### Performance Metrics:
+
+| Metric             | ASIO         | Winsock      | Unix BSD Socket |
+|--------------------|--------------|--------------|----------------|
+| **Latency (1KB)**  | 0.04ms       | 0.04ms       | 0.06ms         |
+| **Latency (1MB)**  | 0.2ms        | 0.4ms        | 0.5ms          |
+| **CPU Usage**      | less than 1% | less than 1% | less than 1%   |
+
+---
```

Comment:
You can add a text to describe the experiment and the result (what did you do and what result did you observe), then draw a conclusion about which library we'll choose and why

Reply:
it's right, i will do it to be more precise for our choice thanks :ok_hand: